openwrt_luci_rpc/openwrt_luci_rpc.py

```diff
@@ -62,21 +64,39 @@ def _determine_if_legacy_version(self):
         :return: tuple, with True if legacy and the URL to
                 use to lookup devices
         """
+
+        # NEW METHOD TO DETERMINE OPENWRT VERSION EXACTLY
+        # TODO: Check as non-root user
+
+        log.info("getting openwrt version")
+
+        rcp_sys_version_call = Constants.\
+            LUCI_RPC_SYS_PATH.format(self.host_api_url), "exec"
+
+        try:
+            content = self._call_json_rpc(*rcp_sys_version_call,
+                                          "cat /etc/openwrt_version")
+
+            if content is None:
+                raise LuciRpcUnknownError("could not \
+                determine openwrt version")
+
+            self.owrt_version = version.parse(content.strip())
```

Comment:
Good to log the version here. eg

```python
log.info("Determined owrt_version {}".format(self.owrt_version))
```

Reply:
On current releases this is not a valid version. e.g. for me it is getting

`Determined owrt_version r7808-ef686b7292`

```
# cat /etc/openwrt_version
r7808-ef686b7292
```

Reply:
I'm sad, I really thought that would have worked.